pkg/commands/config.go

```diff
@@ -139,6 +159,55 @@ func getBaseImage(platform string, bo *options.BuildOptions) build.GetBase {
 	}
 }
 
+// verifier memoizes digest verification results, so that we avoid redundant
+// digest verification.
+type verifier struct {
+	m       sync.Mutex
+	entries map[name.Digest]*entry
+}
+
+// entry is used to hold the sync.Once used to execute the digest verification,
+// and the resulting error (possibly nil), that results from verification.
+type entry struct {
+	once   sync.Once
+	result error
+}
+
+func (v *verifier) verify(ctx context.Context, digest name.Digest) error {
+	e := v.getEntry(digest)
+	e.once.Do(func() {
+		co := &cosign.CheckOpts{
+			RegistryClientOpts: []remote.Option{
+				remote.WithAuthFromKeychain(authn.DefaultKeychain),
+				remote.WithContext(ctx),
+			},
+			ClaimVerifier: cosign.SimpleClaimVerifier,
+			RekorURL:      "https://rekor.sigstore.dev",
+			RootCerts:     fulcioroots.Get(),
+			SignatureRepo: digest.Repository,
+		}
+
+		_, e.result = cosign.Verify(ctx, digest, co)
+		if e.result == nil {
+			log.Printf("Verified %s", digest)
+		}
+	})
+	return e.result
+}
+
+func (v *verifier) getEntry(digest name.Digest) *entry {
+	v.m.Lock()
+	defer v.m.Unlock()
+	// See if there's already an entry for this digest.
+	if e, ok := v.entries[digest]; ok {
```

Comment:
nit: You could hide `v.entries` initialization in here, instead of making it the responsibility of the `verifier` instantiator, above.